real_time/arduino_src/radio_buggy_mega/fingerprint.py

```diff
@@ -0,0 +1,51 @@
+""" fingerprint.py: 
+* Description: This is a python script to create a .h file defining the time 
+*   of compilation and who compiled.  Also details whether files had been
+*   modified since the last commit
+*
+*
+*   By Sean Buckley
+"""
+
+
+import os
+import datetime
+import socket #Used for getting hostname
+import subprocess
+
+def main():
+    print("Now creating fingerprint.h...\n")
+    file = open("fingerprint.h", 'w')
+    file.write("/* fingerprint.h: A compilation-time created file containing compilation time, \n")
+    file.write("closest version of git, and other information of */\n\n")
+    time = datetime.datetime.now()
+    file.write("#define COMPDATE \"%02d:%02d:%04d\" //day:month:year\n" %(time.day, time.month, time.year))
+    file.write("#define COMPTIME \"%02d:%02d:%02d\" //hour:minute:second\n" %(time.hour, time.minute, time.second))
+    file.write("#define HOSTNAME \"%s\"\n" %socket.gethostname().split(".")[0])
+
+
+    result = str(subprocess.check_output(["git", "status"]))
+
+    prev = ""
+    branch = ""
+    clean = False
+
+    for string in result.split():
+        if (prev == "branch"):
+            branch = string
+        if (string == "clean"):
+            clean = True
```

Comment:
We can also get this info using the git plumbing (low level) layer by looking at the return code of git diff-index:
```
02:25] [ihartwig@nilla2:radio_buggy_mega]% git diff-index --quiet HEAD
[02:25] [ihartwig@nilla2:radio_buggy_mega]% echo $?
1
[02:25] [ihartwig@nilla2:radio_buggy_mega]% git diff-index HEAD
:100755 100755 6ec4e792ddbcee9e4aad7d671e05313edbcf31b8 0000000000000000000000000000000000000000 M      real_time/arduino_src/radio_buggy_mega/Makefile
[02:25] [ihartwig@nilla2:radio_buggy_mega]% echo $?
0
```
If there are changes, you should get 1 as a return code. No changes should give 0. Apparently the return code is only useful with the --quiet flag though.